code/mob/living/carbon/human.dm

```diff
@@ -1139,13 +1139,26 @@
 					src.set_a_intent(INTENT_DISARM)
 				return
 		else
-			if (src.client.check_key(KEY_THROW) && src.a_intent == "help" && isliving(target) && BOUNDS_DIST(src, target) <= 0)
-				var/obj/item/thing = src.equipped() || src.l_hand || src.r_hand
-				if (thing)
-					usr = src
-					var/mob/living/living_target = target
-					living_target.give_item()
-					return
+			if (src.client.check_key(KEY_THROW) && BOUNDS_DIST(src, target) <= 0)
+				if (!src.client.mob.equipped())
+					var/turf/T = get_turf(target)
+					var/list/items = list()
+					var/obj/item/I
+					for (var/i = 1 to min(10, length(T.contents)))
+						I = T.contents[i]
+						if (istype(I) && !I.anchored)
+							items += I
+					if (length(items))
+						I = items[length(items)] // top-most item is listed last in content
+						I.attack_hand(src.client.mob)
```

Comment:
`attack_hand()` should never be called directly. Use `AttackHand()`, or perhaps `interact_verb()`, which calls `client.Click()` on the atom